src/jit/codegencommon.cpp

```diff
@@ -11221,10 +11221,7 @@ bool Compiler::IsHfa(GenTreePtr tree)
 var_types Compiler::GetHfaType(GenTreePtr tree)
 {
 #ifdef FEATURE_HFA
-    if (tree->TypeGet() == TYP_STRUCT)
-    {
-        return GetHfaType(gtGetStructHandleIfPresent(tree));
-    }
+    return GetHfaType(gtGetStructHandleIfPresent(tree));
```

Comment:
`gtGetStructHandleIfPresent()` already has a check for `varTypeIsStruct()` so simply eliminate the overly narrow check.